src/lightning/fabric/strategies/xla.py

```diff
@@ -46,6 +46,7 @@ def __init__(
         parallel_devices: Optional[List[torch.device]] = None,
         checkpoint_io: Optional[CheckpointIO] = None,
         precision: Optional[Precision] = None,
+        broadcast_master_params: bool = True,
```

Comment:
Since lightning has the `seed_everything` utility, we could check where [`PL_GLOBAL_SEED`](https://github.com/Lightning-AI/lightning/blob/9eedf7ae4b5cd398931af14ca7eac0605ef740ee/src/lightning/fabric/utilities/seed.py#L54) is available in the environment and that will tell us if we need to call `broadcast_master_params` and avoid this flag. What do you think?

Reply:
I had this thought too but have one concern. What happens if `seed_everything` is defined on each device with a different seed? For example: [Lit-Llama](https://github.com/Lightning-AI/lit-llama/blob/main/train_shakespeare.py#LL49C18-L49C18) defines the seed based on the device's rank `fabric.seed_everything(1337 + fabric.global_rank)`. In this case, would we need to call `broadcast_master_params`? If so, checking whether `PL_GLOBAL_SEED` exists in the environment would not be sufficient.

Reply:
Good point. Normally we recommend using the same seed across ranks if the DistributedSampler is used, but that script doesn't use dataloaders at all.
cc @awaelchli

How much overhead does this add? Because ordinary DDP always does this

Reply:
@gkroiz @carmocca Yes correct, it really depends on the use case. Lit-llama defines it this way to make sure we don't end up with the same data samples on each rank. One could have also used the torch utils DataLoader and DistributedSampler pattern as Carlos said. Two different ways, same purpose. For simplicity there, it was chosen not to introduce the dataloader pattern.

Regarding broadcasting the parameters. DDP does it by default yes, while FSDP exposes it as an argument `sync_module_states` (https://pytorch.org/docs/stable/fsdp.html?highlight=fsdp#module-torch.distributed.fsdp) that is False by default. I would say it is fine to leave this decision up to the strategy. I would rather not tie it to the global seed setting, since this is quite a silent behavior that the user wouldn't be able to override.